src/main/java/org/datadog/jenkins/plugins/datadog/traces/StepDataAction.java

```diff
@@ -0,0 +1,28 @@
+package org.datadog.jenkins.plugins.datadog.traces;
+
+import hudson.model.InvisibleAction;
+import org.datadog.jenkins.plugins.datadog.model.StepData;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Keeps the Step data during a certain Run.
+ */
+public class StepDataAction extends InvisibleAction implements Serializable {
```

Comment:
This class datasize seems to grow unbounded?  Previously, there was a `remove` method which appeared to keep the data size in check -- will this object go out of scope and be garbage collected at some point?

Reply:
The Action is bound to the Run instance. I assume that Jenkins will destroy all objects associated with that Run instance once this has finished.

Reply:
Ok, that makes sense.